modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/editor/BalloonEditor.js

```diff
@@ -98,6 +107,7 @@ const BalloonEditor = ({config = {}, contents, name, ...otherProps}) => {
 				});
 			}}
 			type="inline"
+			{...imageDirectionShim}
```

Comment:
I added as a style because I tried to pass className or take a look in what can be inserted using CKEditor's API but I didn't found anything :(

Reply:
*Shim not sounds good. Does someone have a suggestion?

Reply:
It's always better to add a class than to set inline styles. Something like this will work:
```js
onInstanceReady={(event) => {
	const editable = editor.editable();

	editable.attachClass('lfr-editable');
...
```

I am not sure if there is a cleaner way to set editable container class. Looking at [implementation of editable](https://github.com/ckeditor/ckeditor4/blob/4.16.1/core/editable.js#L884-L894), I don't think there is, but there's a lot of code in there :frowning:

Reply:
```js
editable.attachClass('overflow-auto');
```

worked well with `overflow-auto`. It relies in clay/bs utilities, should I create another class for it or does make sense just adding it?

Reply:
See https://github.com/liferay-frontend/liferay-portal/pull/1339/commits/df033efedceb335c9f7712859befaa2bb5003145

Reply:
> worked well with overflow-auto. It relies in clay/bs utilities, should I create another class for it or does make sense just adding 

I personally don't like CSS classes that are 1:1 with styles:
1. Using them tends to lead to classes bloat.
2. You cannot fine-tune styles priorities with class hierarchy, the way you can with styles in a CSS file. I guess the counter-argument is that you are avoiding complexities that can easily crop up with those kinds of fine-tuning.
3. I like it when all or majority styles are in one place. This is kind of a moot point, since CSS bleeds so much it's pointless to look just in a file. You need to see it on page to see how it looks and what is applied.

In the end, it's a tradeoff and a matter of personal preference. If you prefer it this way, :+1: 

Quick ping @edalgrin @marcoscv-work . Can we rely on clay/bs utilities?

Reply:
It depends on the context:
- if we already have some custom CSS we should extend it and add the modifier to include this case.
- If we need to load clay-css just for this class, we shouldn't!
- If we are already using clay-css and we don't need to provide a customizable product (example: a theme) we can use `overflow-auto`.

About the "a single class for a single CSS properties" it's up to you. Tailwind is built using a very similar concept. I personally think it depends on the project, it can be very useful in some but too chaotic in others 🤷

Reply:
> If we need to load clay-css just for this class, we shouldn't!

Hmm that's a good point. We don't have `@clayui/css` as a dependency in `frontend-editor-ckeditor-web`. I guess this works only because something else loads it, probably theme? This is not a major deal, as balloon editor is not meant to be used outside DXP. But still, there is no need set a fragile non-explicit dependency, if we don't need to. A safer and not very painful alternative is to have the `liferay-editable` class, and a style entry in `main.scss`.

Reply:
Thanks for bringing this context Edu! I was needing more clearance and "If we are already using clay-css and we don't need to provide a customizable product (example: a theme) we can use overflow-auto." helped :)

> I guess this works only because something else loads it, probably theme?

Yes, for this use case. For others, like tabletoolbar, where [We are adding a `CKEDITOR.skin.getSkinPath`](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/plugins/tabletoolbar/plugin.js#L1104), ClayCSS(or a part of it) is being loaded via `moono-lexicon`.

Reply:
> But still, there is no need set a fragile non-explicit dependency, if we don't need to. A safer and not very painful alternative is to have the liferay-editable class, and a style entry in main.scss

Makes sense, I'll add this class. I don't want to rely too much in theme styles :)

Reply:
See 26ed34f39269f34274f716c6ac3c4effdf83c014

Reply:
In https://github.com/liferay-frontend/liferay-portal/commit/26ed34f39269f34274f716c6ac3c4effdf83c014 , I named the css class as "lfr-editable" but sometimes this green border is being applied. Taking a look in DXP, there is definitions using this same naming. I'll go with `liferay-editable` instead. Force-pushing...

<img width="1006" alt="Screen Shot 2021-08-11 at 14 08 49" src="https://user-images.githubusercontent.com/7663513/129072822-dc69c93b-0f5a-4bb3-9c57-ad45214279f1.png">